src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -1296,5 +1296,15 @@ public static int GetLastWin32Error()
         {
             return GetLastPInvokeError();
         }
+
+        /// <summary>
+        /// Gets the system error message for the supplied error code.
+        /// </summary>
+        /// <param name="error">The error code.</param>
+        /// <returns>The error message associated with <paramref name="error"/>.</returns>
+        public static string GetPInvokeErrorMessage(int error)
+        {
+            return Win32Exception.GetErrorMessage(error);
```

Comment:
I'd prefer to invert the relationship, having the implementation of GetErrorMessage be here and Win32Exception call GetPInvokeErrorMessage.

Reply:
I agree that makes more sense, and Marshal is already in Windows/Unix halves too.

Static members on exception types feel odd to me, and I'd expect it to be on Marshal as there are several similar API there already.

Reply:
Plus, it helps keep the term "Win32Exception" out of customer code that's not Windows specific.